terraform/datadog_email_activity.tf

```diff
@@ -0,0 +1,425 @@
+      sources              = ["eventType"]
+      source_type          = "attribute"
+      target               = "event-type"
+      target_type          = "tag"
+      target_format        = "string"
+      preserve_source      = true
+      override_on_conflict = false
+      name                 = "Map 'eventType' to 'event-type' tag"
+      is_enabled           = true
+    }
+  }
+  processor {
+    attribute_remapper {
+      sources              = ["mail.tags.user_role"]
+      source_type          = "attribute"
+      target               = "user-role"
+      target_type          = "tag"
+      target_format        = "string"
+      preserve_source      = true
+      override_on_conflict = false
+      name                 = "Map mail.tags.user_role to 'user-role' tag"
+      is_enabled           = true
+    }
+  }
+  processor {
+    attribute_remapper {
+      sources              = ["mail.tags.team_id"]
+      source_type          = "attribute"
+      target               = "team-id"
+      target_type          = "tag"
+      target_format        = "string"
+      preserve_source      = true
+      override_on_conflict = false
+      name                 = "Map mail.tags.team_id to 'team-id' tag"
+      is_enabled           = true
+    }
+  }
+  processor {
+    attribute_remapper {
+      sources              = ["mail.tags.organization_id"]
+      source_type          = "attribute"
+      target               = "organization-id"
+      target_type          = "tag"
+      target_format        = "string"
+      preserve_source      = true
+      override_on_conflict = false
+      name                 = "Map mail.tags.organization_id to 'organization-id' tag"
+      is_enabled           = true
+    }
+  }
+  processor {
+    attribute_remapper {
+      sources              = ["mail.tags.ses:configuration-set"]
+      source_type          = "attribute"
+      target               = "configuration-set"
+      target_type          = "tag"
+      target_format        = "string"
+      preserve_source      = true
+      override_on_conflict = false
+      name                 = "Map mail.tags.ses:configuration-set to 'configuration-set' tag"
+      is_enabled           = true
+    }
+  }
+  processor {
+    attribute_remapper {
+      sources              = ["mail.tags.version"]
+      source_type          = "attribute"
+      target               = "version"
+      target_type          = "tag"
+      target_format        = "string"
+      preserve_source      = true
+      override_on_conflict = false
+      name                 = "Map mail.tags.version to 'version' tag"
+      is_enabled           = true
+    }
+  }
+  processor {
+    trace_id_remapper {
+      sources    = ["mail.tags.dd_trace_id, mail.tags.dd_span_id"]
+      name       = "trace id remapper"
+      is_enabled = true
+    }
+  }
+}
+
+resource "datadog_logs_metric" "gost_ses_email_sending_event_subscription" {
+  name = "gost_ses_email_sending_event_subscription"
```

Comment:
@mayabose I assume this is related to your [comment](https://github.com/usdigitalresponse/usdr-gost/pull/3387#issuecomment-2294206146),
> It looks like terraform doesn't want any periods in the names of the metrics.

I believe the [error message](https://github.com/usdigitalresponse/usdr-gost/actions/runs/10423821751/job/28871460690#step:14:265) you're seeing in the `terraform plan` output
```
A name must start with a letter or underscore and may contain only letters,
digits, underscores, and dashes.
```
is referring to the name of the _resource_, not the `name` attribute of the metric itself. Terraform resource names (e.g. `gost_ses_sending_event_subscription`) cannot have `.` characters because it incorrectly formats the full resource address when referencing elsewhere (e.g. `datadog_logs_metric.gost_ses_email_sending_event_subscription`). However, the `name` attribute here doesn't need to match the resource name exactly.

I would suggest keeping the resource name as-is, while re-adding `.` characters to the name of the metric that will be created in Datadog:
```suggestion
resource "datadog_logs_metric" "gost_ses_email_sending_event_subscription" {
  name = "gost.ses.email_sending_event.subscription"
```